llrt_core/src/modules/timers.rs

```diff
@@ -15,103 +17,143 @@ use rquickjs::{
     prelude::{Func, Opt},
     qjs, CatchResultExt, Ctx, Function, Persistent, Result,
 };
+use tokio::{
+    select,
+    sync::Notify,
+    time::{Instant, Sleep},
+};
 
 use crate::{module_builder::ModuleInfo, modules::module::export_default, vm::Vm};
 
 static TIMER_ID: AtomicUsize = AtomicUsize::new(0);
-static TIME_POLL_ACTIVE: AtomicBool = AtomicBool::new(false);
-pub struct TimeoutRef {
+static RT_TIMER_STATE: Lazy<Mutex<Vec<RuntimeTimerState>>> = Lazy::new(|| Mutex::new(Vec::new()));
+
+pub struct RuntimeTimerState {
+    timers: Vec<Timeout>,
+    rt: *mut qjs::JSRuntime,
```

Comment:
Since this is runtime agnostic do you want to move it to modules? I can also do it for you later.